src/runtime/lpython_builtin.py

```diff
@@ -743,6 +743,17 @@ def _lpython_str_find(s: str, sub: str) -> i32:
 
     return res
 
+@overload
+def _lpython_list_count(l: list[i32], x: i32) -> i32:
+    count : i32
+    count = 0
+    i : i32
+    i = 0
+    for i in l:
+        if(i == x):
+            count = count + 1
+    return count
```

Comment:
Well three approaches,

1. Use generics - Very messy and not worth it at the moment.
2. Use IntrinsicFunction infrastructure to programmatically generate the code for these functions and call it in `intrinsic_functions.cpp` pass.
3. Insert a new node, `ListCount(expr list, expr x)` in ASR.asdl and implement it directly in LLVM backend.

Approach 3 is most doable right now and will generate the most efficient LLVM code. However to scale well across all backends, approach 2 is the best one but should be done later. There is no disadvantage in going for approach 3. Once approach 2 is implemented we can remove `ListCount` node easily. @certik What do you say?

Reply:
Yes, 3 is fine, later we'll move to 2.